design/one-pager-pluggable-secret-stores.md

```diff
@@ -154,47 +153,29 @@ message DeleteOptions {
 
 A _Secret Store Plugin_:
 
-- Will be a binary packaged as a container image.
-- Should run a gRPC server listening at Unix domain socket.
-- May support one or more Secret Stores.
+- Will be a k8s deployment behind a service.
+- Should run a gRPC server.
+- Should support one Secret Store.
 - Should deploy CRD(s) for supported store configurations at startup.
-- Will run as a sidecar next to Crossplane and Provider containers.
+- Should have server certificates signed by the kubernetes cluster root CA.
+- Should require and verify client certificates.
 
-#### Plugin Binary vs Secret Store Implementation
+### Deployment
 
-All `rpc` methods have `ConfigReference` as their first parameter. So,
-one Secret Store Plugin may support more than one Secret Store implementation,
-e.g. Vault, AWS, GCP, Azure ... all in a single Plugin binary, and related
-implementation could be called depending on a given config similar to how it is
-being done in [RuntimeStoreBuilder] today. Implementation of individual Secret
-Stores still need to satisfy the same [Store] interface today and could live in
-separate repositories. However, they would need to be packaged and served by a
-Secret Store Plugin binary to be used with Crossplane.
+Deployment of a Secret Store Plugin will be similar to the current provider 
+deployment with one exception. We need to generate a server certificate 
+during deployment time and mount it to the plugin pod. This is required to enable
+mTLS communication.
 
-### Deployment
+In addition to the server certificates, we will also generate a client certificate
+for each provider and store it in a k8s secret. These certificates will be
+generated by package manager during the package installation. 
+
+Since Crossplane itself will be a gRPC client, we need to generate and mount
```

Comment:
nit:

> Since the core Crossplane communicates with the secret stores as well, it will also act as a gRPC client to communicate with the plugin. Hence we need to generate ...